server/utils/settingspanel/handler.go

```diff
@@ -29,33 +30,37 @@ func Init(h *httputils.Handler, panel Panel) {
 func (sh *handler) handleAction(w http.ResponseWriter, r *http.Request) {
 	mattermostUserID := r.Header.Get("Mattermost-User-ID")
 	if mattermostUserID == "" {
-		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		utils.SlackAttachmentError(w, "Error: Not authorized.")
 		return
 	}
 
 	request := model.PostActionIntegrationRequestFromJson(r.Body)
 	if request == nil {
-		http.Error(w, "invalid request", http.StatusBadRequest)
+		utils.SlackAttachmentError(w, "Error: invalid request")
 		return
 	}
 
 	id, ok := request.Context[ContextIDKey]
 	if !ok {
-		http.Error(w, "invalid request", http.StatusBadRequest)
+		utils.SlackAttachmentError(w, "Error: no id in context")
 		return
 	}
 
 	value, ok := request.Context[ContextButtonValueKey]
 	if !ok {
 		value, ok = request.Context[ContextOptionValueKey]
 		if !ok {
-			http.Error(w, "valid key not found", http.StatusBadRequest)
+			utils.SlackAttachmentError(w, "Error: valid key not found")
 			return
 		}
 	}
 
 	idString := id.(string)
-	sh.panel.Set(mattermostUserID, idString, value)
+	err := sh.panel.Set(mattermostUserID, idString, value)
+	if err != nil {
+		utils.SlackAttachmentError(w, "Error: cannot set the property, "+err.Error())
```

Comment:
Do these errors get shown in the UI? It seems a bit low-level

Reply:
Yes, these would get shown in the UI. My idea is the error to be meaningful to debug, but we could move the internals of the error to a log in the server. Nevertheless, that would imply passing a logger to the handler, which will make things a bit dirtier.